aws/resource_aws_xray_sampling_rule.go

```diff
@@ -0,0 +1,212 @@
+)
+
+func resourceAwsXraySamplingRule() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsXraySamplingRuleCreate,
+		Read:   resourceAwsXraySamplingRuleRead,
+		Update: resourceAwsXraySamplingRuleUpdate,
+		Delete: resourceAwsXraySamplingRuleDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"rule_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"resource_arn": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"priority": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"fixed_rate": {
+				Type:     schema.TypeFloat,
+				Required: true,
+			},
+			"reservoir_size": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"service_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"service_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"host": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"http_method": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"url_path": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"version": {
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"attributes": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+			"rule_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsXraySamplingRuleCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).xrayconn
+	samplingRule := &xray.SamplingRule{
+		RuleName:      aws.String(d.Get("rule_name").(string)),
+		ResourceARN:   aws.String(d.Get("resource_arn").(string)),
+		Priority:      aws.Int64(int64(d.Get("priority").(int))),
+		FixedRate:     aws.Float64(d.Get("fixed_rate").(float64)),
+		ReservoirSize: aws.Int64(int64(d.Get("reservoir_size").(int))),
+		ServiceName:   aws.String(d.Get("service_name").(string)),
+		ServiceType:   aws.String(d.Get("service_type").(string)),
+		Host:          aws.String(d.Get("host").(string)),
+		HTTPMethod:    aws.String(d.Get("http_method").(string)),
+		URLPath:       aws.String(d.Get("url_path").(string)),
+		Version:       aws.Int64(int64(d.Get("version").(int))),
+	}
+
+	if v, ok := d.GetOk("attributes"); ok {
+		samplingRule.Attributes = stringMapToPointers(v.(map[string]interface{}))
+	}
+
+	params := &xray.CreateSamplingRuleInput{
+		SamplingRule: samplingRule,
+	}
+
+	out, err := conn.CreateSamplingRule(params)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(*out.SamplingRuleRecord.SamplingRule.RuleName)
+
+	return resourceAwsXraySamplingRuleRead(d, meta)
+}
+
+func resourceAwsXraySamplingRuleRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).xrayconn
+	params := &xray.GetSamplingRulesInput{}
+
+	for {
```

Comment:
This seems like it should be simplified to use the `getSamplingRule` function (I have some additional notes associated with that function as well below). 😄 e.g.

```go
samplingRule, err := getXraySamplingRule(conn, d.Id())

if err != nil {
	return fmt.Errorf("error reading XRay Sampling Rule (%s): %s", d.Id(), err)
}

if samplingRule == nil {
	log.Printf("[WARN] XRay Sampling Rule (%s) not found, removing from state", d.Id())
	d.SetId("")
	return nil
}

d.Set("rule_name", samplingRule.RuleName)
// ... set other attributes ...
```